test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTextAreaTest.cs

```diff
@@ -40,6 +42,125 @@ public void TextAreaFor_DoesNotGeneratePlaceholderAttribute_WhenNoPlaceholderPre
             Assert.DoesNotContain(@"placeholder=""HtmlEncode[[placeholder]]""", result, StringComparison.Ordinal);
         }
 
+        public static TheoryData TextAreaFor_UsesModelValueForComplexExpressionsData
+        {
+            get
+            {
+                return new TheoryData<Expression<Func<ComplexModel, string>>, string>
+                {
+                    {
+                        model => model.Property3["key"],
+                        "<textarea id=\"HtmlEncode[[pre_Property3_key_]]\" name=\"HtmlEncode[[pre.Property3[key]]]\">" + Environment.NewLine +
+                        "HtmlEncode[[Prop3Val]]</textarea>"
+                    },
+                    {
+                        model => model.Property4.Property5,
+                        "<textarea id=\"HtmlEncode[[pre_Property4_Property5]]\" name=\"HtmlEncode[[pre.Property4.Property5]]\">" + Environment.NewLine +
+                        "HtmlEncode[[Prop5Val]]</textarea>"
+                    },
+                    {
+                        model => model.Property4.Property6[0],
+                        "<textarea id=\"HtmlEncode[[pre_Property4_Property6_0_]]\" name=\"HtmlEncode[[pre.Property4.Property6[0]]]\">" + Environment.NewLine +
+                        "HtmlEncode[[Prop6Val]]</textarea>"
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TextAreaFor_UsesModelValueForComplexExpressionsData))]
+        public void TextAreaFor_ComplexExpressions_UsesModelValue(
```

Comment:
Not a big deal but oddly consistent. Why isn't this named `TextAreaFor_UsesModelValueForComplexExpressions()`? Or, why isn't the data named `TextAreaFor_ComplexExpressions_UsesModelValueData`?